Jellyfin.Api/Controllers/NotificationsController.cs

```diff
@@ -0,0 +1,145 @@
+#nullable enable
+#pragma warning disable CA1801
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using Jellyfin.Api.Models.NotificationDtos;
+using MediaBrowser.Controller.Library;
+using MediaBrowser.Controller.Notifications;
+using MediaBrowser.Model.Dto;
+using MediaBrowser.Model.Notifications;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Jellyfin.Api.Controllers
+{
+    /// <summary>
+    /// The notification controller.
+    /// </summary>
+    public class NotificationsController : BaseJellyfinApiController
+    {
+        private readonly INotificationManager _notificationManager;
+        private readonly IUserManager _userManager;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="NotificationsController" /> class.
+        /// </summary>
+        /// <param name="notificationManager">The notification manager.</param>
+        /// <param name="userManager">The user manager.</param>
+        public NotificationsController(INotificationManager notificationManager, IUserManager userManager)
+        {
+            _notificationManager = notificationManager;
+            _userManager = userManager;
+        }
+
+        /// <summary>
+        /// Endpoint for getting a user's notifications.
+        /// </summary>
+        /// <param name="userId">The user's ID.</param>
+        /// <param name="isRead">An optional filter by IsRead.</param>
+        /// <param name="startIndex">The optional index to start at. All notifications with a lower index will be dropped from the results.</param>
+        /// <param name="limit">An optional limit on the number of notifications returned.</param>
+        /// <returns>A read-only list of all of the user's notifications.</returns>
+        [HttpGet("{UserID}")]
+        [ProducesResponseType(typeof(NotificationResultDto), StatusCodes.Status200OK)]
+        public NotificationResultDto GetNotifications(
+            [FromRoute] string userId,
+            [FromQuery] bool? isRead,
+            [FromQuery] int? startIndex,
+            [FromQuery] int? limit)
+        {
+            return new NotificationResultDto();
```

Comment:
Yeah. This service is actually unused(at least in the official web client), and was only half implemented. That's kind of why I picked it for my first one to move, so that if anything went wrong it wouldn't matter as much.